path/error.go

```diff
@@ -1,25 +1,32 @@
 package path
 
 import (
+	"errors"
 	"fmt"
 )
 
+var (
+	ErrExpectedImmutable      = errors.New("path was expected to be immutable")
+	ErrInsufficientComponents = errors.New("path does not have enough components")
+	ErrUnknownNamespace       = errors.New("unknown namespace")
+)
+
 type ErrInvalidPath struct {
-	error error
-	path  string
+	err  error
+	path string
 }
 
 func (e ErrInvalidPath) Error() string {
-	return fmt.Sprintf("invalid path %q: %s", e.path, e.error)
+	return fmt.Sprintf("invalid path %q: %s", e.path, e.err)
 }
 
 func (e ErrInvalidPath) Unwrap() error {
-	return e.error
+	return e.err
 }
 
 func (e ErrInvalidPath) Is(err error) bool {
 	switch err.(type) {
-	case ErrInvalidPath:
+	case ErrInvalidPath, *ErrInvalidPath:
```

Comment:
Is this necessary? Seems like this isn't unique to this case and happens everywhere there is an `errors.Is` check, but I don't think doing this double check is that standard. There might be linters that check for this error as IIRC I've seen Goland (Jetbrains IDE) complain if you use the wrong error type (with or without a pointer) when calling `.Is`